src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/Rules/Operations/FormattingOperations.cs

```diff
@@ -208,7 +208,7 @@ internal static IEnumerable<AlignTokensOperation> GetAlignTokensOperations(IEnum
         /// <summary>
         /// return AdjustNewLinesOperation for the node provided by the given formatting rules
         /// </summary>
-        internal static AdjustNewLinesOperation GetAdjustNewLinesOperation(IEnumerable<FormattingRule> formattingRules, SyntaxToken previousToken, SyntaxToken currentToken, AnalyzerConfigOptions options)
+        internal static AdjustNewLinesOperation? GetAdjustNewLinesOperation(IEnumerable<FormattingRule> formattingRules, SyntaxToken previousToken, SyntaxToken currentToken, AnalyzerConfigOptions options)
```

Comment:
💭 What about having a default instance `AdjustNewLinesOperation.None` or something to indicate no changes are needed without actually having the nullable returns?

Reply:
i've considered it... but it has challenges.  right now with nullable, consumers know if they ahve a good value or not, and the compiler prevents mistakes.  i.e. if someone calls this, they can't just take the value and do `.Line` on it.  They *must* check it for null and decide what to do.

With `None` sentinels things get much trickier, it's unclear who should be checking the value or when.  I would be worried about a .None slipping around causing problems as people blindly use it without realizing that's a bad thing.